airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/FileSystemConfigPersistence.java

```diff
@@ -58,18 +59,25 @@
   // root for where configs are stored
   private final Path configRoot;
 
-  public static ConfigPersistence createWithValidation(final Path storageRoot) throws InterruptedException {
-    LOGGER.info("Constructing file system config persistence (root: {})", storageRoot);
-
-    Path configRoot = storageRoot.resolve(CONFIG_DIR);
-    int totalWaitingSeconds = 0;
-    while (!Files.exists(configRoot)) {
-      LOGGER.warn("Config volume is not ready yet (waiting time: {} s)", totalWaitingSeconds);
-      Thread.sleep(INTERVAL_WAITING_SECONDS * 1000);
-      totalWaitingSeconds += INTERVAL_WAITING_SECONDS;
+  /**
+   * Check if there are existing configs under the storage root. Previously the seed container copies
+   * the configs to the storage root, it may take some time for the operation to complete and for the
+   * CONFIG_DIR to show up. So we cannot infer anything based on the existence of this directory. Now
+   * this seed generation step has been removed. So we can tell immediately whether CONFIG_DIR exists
+   * or not. If CONFIG_DIR exists, it means the user has just migrated Airbyte from an old version
```

Comment:
how do we know they "just" migrated? does the directory get deleted?

Reply:
Yes. This config directory will not be created in newer Airbyte versions, as the copy step is removed in `seed.Dockerfile`:

```diff
- COPY build/config_init/resources/main/config seed/config 
```

So only the old users will see it.